Core/include/Acts/Vertexing/GaussianTrackDensity.ipp

```diff
@@ -0,0 +1,162 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <math.h>
+
+template <typename input_track_t>
+std::pair<double, double>
+Acts::GaussianTrackDensity<input_track_t>::globalMaximumWithWidth(
+    State& state, const std::vector<const input_track_t*>& trackList,
+    const std::function<BoundParameters(input_track_t)>& extractParameters)
+    const {
+  addTracks(state, trackList, extractParameters);
+
+  double maxPosition = 0.;
+  double maxDensity = 0.;
+  double maxSecondDerivative = 0.;
+
+  for (const auto& track : state.trackEntries) {
+    double trialZ = track.z;
+    double density = 0.;
+    double firstDerivative = 0.;
+    double secondDerivative = 0.;
+    density = trackDensity(state, trialZ, firstDerivative, secondDerivative);
+    if (secondDerivative >= 0. || density <= 0.) {
+      continue;
+    }
+    updateMaximum(trialZ, density, secondDerivative, maxPosition, maxDensity,
```

Comment:
Again, this is addressed in the header: 
```cpp
/// @param[out] maxZ The max z value
/// @param[out] maxValue The max value at z position
/// @param[out] maxSecondDerivative The max second derivative
```
But I have changed it now.

Reply:
Excellent.

Reply:
Do you still want me to change it to ```std::array<double, 3>``` as outputs everywhere without going with the references? I'm working on this since yesterday (actually it was only 2 min of work) but for some reason one of my unit tests fails afterwards which is super weird. I would prefer leaving it like this with references while making sure that it is made very obvious that they are being altered there.

Reply:
Changed as well.